pkgs/development/beam-modules/build-rebar3.nix

```diff
@@ -52,29 +49,19 @@ let
       rm -f rebar rebar3
     '' + postPatch;
 
-    configurePhase = ''
-      runHook preConfigure
-      ${erlang}/bin/escript ${rebar3.bootstrapper} ${debugInfoFlag}
-      runHook postConfigure
-    '';
-
     buildPhase = ''
       runHook preBuild
-      HOME=. rebar3 compile
-      ${if compilePorts then ''
-        HOME=. rebar3 pc compile
-      '' else ""}
+      HOME=. rebar3 bare compile \
+        --paths "${toString (map (p: "${p}/lib/erlang/lib/*/ebin") beamDeps)}"
       runHook postBuild
     '';
 
     installPhase = ''
       runHook preInstall
       mkdir -p "$out/lib/erlang/lib/${name}-${version}"
       for reldir in src ebin priv include; do
-        fd="_build/default/lib/${name}/$reldir"
-        [ -d "$fd" ] || continue
-        cp -Hrt "$out/lib/erlang/lib/${name}-${version}" "$fd"
-        success=1
+        [ -d "$reldir" ] || continue
```

Comment:
Just asking a question since I'm not sure of what was happening before and after.

As I understand it
Before
- in the build phase files are compiled to `_build/default/${name}`
- check for the presence of a specific directory (src ebin priv include) inside the build
- if the directory is there, copy the contents of the directory into the convention defined path
After
- check for the presence of (src ebin priv include) directly inside $src
- if the directory is there, copy it's content too the convention defined path

In this pr does this mean that the bare compile will compile directly into $src and not into _build ?
There is one step I'm missing here.

Reply:
Hi, yes you've got this exactly right. AFAIK the only documentation for `rebar3 bare compile` is the source code, there's a parameter called `outdir` but it defaults to the working directory.